packages/cli/src/zosfiles/edit/Edit.utils.ts

```diff
@@ -0,0 +1,227 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, IHandlerParameters, ImperativeError, ProcessUtils, GuiResult,
+    TextUtils, IDiffOptions, RestClientError } from "@zowe/imperative";
+import { tmpdir } from "os";
+import { Download, Upload, IZosFilesResponse } from "@zowe/zos-files-for-zowe-sdk";
+import LocalfileDatasetHandler from "../compare/lf-ds/LocalfileDataset.handler";
+import { CompareBaseHelper } from "../compare/CompareBaseHelper";
+import { CliUtils } from "@zowe/imperative";
+import { unlink, existsSync } from "fs";
+
+export class LocalFile {
+    dir: string;
+    zosResp: IZosFilesResponse;
+}
+export enum Prompt {
+    useStash,
+    doneEditing,
+    continueToUpload
+}
+export class EditUtilities {
+    // Build tmp path
+    public static async buildTmpDir(commandParameters: IHandlerParameters): Promise<string>{
+        const ext = (commandParameters.arguments.extension ? commandParameters.arguments.extension : ".txt");
+        const hashLen = 8;
+        let fileName: string;
+        if (commandParameters.positionals.includes('uss')){
+            // Hash in a repeatable way if uss fileName (to get around any potential special characters in name)
+            fileName = commandParameters.arguments.file;
+            const crypto = require("crypto");
+            const hash = crypto.createHash('sha1').update(fileName).digest('hex').substring(0,hashLen);
```

Comment:
What is our confidence that the 1st eight characters of the digest will be unique among any other files that might be placed in your temp directory? This may not be a problem – just asking.

Reply:
![image](https://user-images.githubusercontent.com/112635587/230202363-254be8d1-6b38-41d4-90cc-fe12b13f74b9.png)
I'm not sure if this is a good answer but the above image shows that the hashing is repeatable and very different even when file names differ just by 1 change

Reply:
Oh I actually reread this and realized I read it wrong initially. I don't have an actual confidence level. Maybe prepending 'zoweTemp' to the hash will be good enough to prevent any collisions?  So `C:\Users\user\AppData\Local\Temp\08deae8d.c` becomes `C:\Users\user\AppData\Local\Temp\zoweTemp08deae8d.c`

Reply:
@ATorrise You might be able to leverage `sha256` instead of `sha1` here, as I believe no collisions have been found with that algorithm yet.

Reply:
ty trae!